### PR TITLE
feat: sytlize markdown inline-tag

### PR DESCRIPTION
**Pull request by trydofor**
stylize markdown enhancement plugin.

enhance tokens like `<tag>keyword</tag>` 

* add/join `tag` attributes, eg: class, style,
* replace `keyword` with others, eg: `NOT` to `NOT🚫`

global config, global enhance. 
also can disable by frontmatter

### CODE REVIEW

File: packages/md-enhance/__tests__/unit/stylize.spec.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  };
+  const markdownIt = MarkdownIt({ linkify: true }).use(stylize, options);
+
+  it("Should render MUST", () => {
+    expect(markdownIt.render(`**MUST**`)).toEqual(
+      '<p><strong class="badge tip">MUST</strong></p>\n'
+    );
+    expect(markdownIt.render(`*MUST*`)).toEqual(
+      '<p><em class="badge tip">MUST</em></p>\n'
+    );
```
Comment (Mister-Hope) [inline packages/md-enhance/__tests__/unit/stylize.spec.ts:36]: Other tests need to added here:

LINE TEST:

```md
**MUST** at the begining of the line.
At the end of the line **MUST**.
```

Should not break other things

```md
`**MUST**` in inline code should be rendered as is.

Other syntax like _italic_ and  **bold** should work with **MUST**.

A invalid syntax like **MUST**and_MUST_ should not be parsed.

Other word not matching keywords like **MUS** and **MUSTS** should not be parsed.
```


Please at lease complete the above tests in your test file.

Comment (Mister-Hope) [reply packages/md-enhance/__tests__/unit/stylize.spec.ts:36]: Also you should add another one: "Plugin without options should not throw"

File: packages/md-enhance/src/node/markdown-it/stylize.ts [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * 其主要目的是使特定词汇具在render后，变得更醒目或生动。
+ * 可以通过front matter设置 noStylize:[t1,t2]禁用t1和t2的增强。
+ * 注意，所有配置项都区分大小写。
+ *
+ * @author trydofor
+ * @since 2022-05-20
+ * @see https://markdown-it.github.io
+ */
+
+import * as Token from "markdown-it/lib/token";
```
Comment (Mister-Hope) [inline packages/md-enhance/src/node/markdown-it/stylize.ts:13]: We would prefer 

```suggestion
import type { default as  Token } from "markdown-it/lib/token";
```

to be constant with other files

File: packages/md-enhance/src/node/markdown-it/stylize.ts [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import { PluginWithOptions } from "markdown-it";
+import { StylizeOption } from "../../shared";
+import { MarkdownEnv } from "@vuepress/markdown";
```
Comment (Mister-Hope) [inline packages/md-enhance/src/node/markdown-it/stylize.ts:16]: Sort them like other files, mark types with `type`

File: packages/md-enhance/src/node/markdown-it/stylize.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+function scan(
```
Comment (Mister-Hope) [inline packages/md-enhance/src/node/markdown-it/stylize.ts:18]: We prefer arrow function code style.

File: packages/md-enhance/src/node/markdown-it/stylize.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    if (repl) {
+      txt.content = repl;
+    }
+  }
+}
+
+export const stylize: PluginWithOptions<StylizeOption> = (md, options) => {
+  if (options == null || Object.keys(options).length == 0) return;
+
+  md.core.ruler.push("stylize-tag", (state) => {
```
Comment (Mister-Hope) [inline packages/md-enhance/src/node/markdown-it/stylize.ts:78]: markdown-it prefer snake_case, and you are welcomed to correct exisiting plugins in new PRs

File: packages/md-enhance/src/node/markdown-it/stylize.ts [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        ? opt.text
+        : "";
+
+    if (repl) {
+      txt.content = repl;
+    }
+  }
+}
+
+export const stylize: PluginWithOptions<StylizeOption> = (md, options) => {
```
Comment (Mister-Hope) [inline packages/md-enhance/src/node/markdown-it/stylize.ts:75]: Nope, you'd better use 
```suggestion
export const stylize: PluginWithOptions<StylizeOption> = (md, options = {} ) => {
```

File: packages/md-enhance/src/shared/stylize.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+     * 标签属性，[attr, value]的数组（二维），如[['class':'badge tip']]。
+     * class和style属性会join，其他属性则覆盖。
+     */
+    attr?: [string, string][];
+
+    /**
+     * truthy时，替换原token.content
+     */
+    text?: string | ((str: string, env: MarkdownEnv) => string);
+  }
```
Comment (Mister-Hope) [inline packages/md-enhance/src/shared/stylize.ts:35]: Make both en and zh in jsdocs like others

File: packages/md-enhance/src/shared/stylize.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  string,
+  {
+    /**
+     * 所在标签，token.tag，如['strong','sup']。
+     */
+    tag: string[];
+
+    /**
+     * 标签属性，[attr, value]的数组（二维），如[['class':'badge tip']]。
+     * class和style属性会join，其他属性则覆盖。
```
Comment (Mister-Hope) [inline packages/md-enhance/src/shared/stylize.ts:27]: Chinese and enligh char should be sperated with single space

**Review by Mister-Hope — CHANGES_REQUESTED** [review]

**Comment by Mister-Hope** [comment]
Besides, welcome to update docs as well

File: packages/md-enhance/__tests__/unit/stylize.spec.ts (hunk 1 above)
Comment (trydofor) [reply packages/md-enhance/__tests__/unit/stylize.spec.ts:36]: `**MUST**and_MUST_` can parsed by markdown-it to `<strong>MUST</strong>and_MUST_`.
so `**MUST**and` is valid syntax

Comment (Mister-Hope) [reply packages/md-enhance/__tests__/unit/stylize.spec.ts:36]: Yes, I realized that, you can change to another test string targeting this.

**Comment by trydofor** [comment]
Thanks for the Theme-Hope, the excellent work :)

**Pull request merged**